From b79759252474ba211aedfbc287d628f520e29cd3 Mon Sep 17 00:00:00 2001
From: Christophe Gragnic <christophegragnic@gmail.com>
Date: Fri, 24 Apr 2015 08:17:00 +0200
Subject: [PATCH 3/5] Emulisp: implement eq0 in int.js.

---
 emulisp/int.js | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/emulisp/int.js b/emulisp/int.js
index e86b35d..aa2005a 100644
--- a/emulisp/int.js
+++ b/emulisp/int.js
@@ -428,6 +428,10 @@ Int.prototype.eq = function (num) {
     return this.cmp(num) === 0;
 };
 
+Int.prototype.eq0 = function () {
+    return this._d.length === 0;
+};
+
 Int.prototype.ne = function (num) {
     return this.cmp(num) !== 0;
 };
-- 
2.1.4

